server_src/runners/history.js

```diff
@@ -364,21 +342,17 @@ class HistoryCollection {
     };
 
     const path = `./images/historyCollection/timelapses/${id}-${fileName}`;
-    if (!fs.existsSync(`./images/historyCollection`)) {
-      fs.mkdirSync(`./images/historyCollection`);
+    if (!fs.existsSync("./images/historyCollection")) {
+      fs.mkdirSync("./images/historyCollection");
     }
-    if (!fs.existsSync(`./images/historyCollection/timelapses`)) {
-      fs.mkdirSync(`./images/historyCollection/timelapses`);
+    if (!fs.existsSync("./images/historyCollection/timelapses")) {
+      fs.mkdirSync("./images/historyCollection/timelapses");
     }
 
     download(url, path, () => {
       logger.info("Downloaded: ", url);
       logger.info(`images/historyCollection/timelapses/${id}-${fileName}`);
-      if (
-        typeof serverSettings.history !== "undefined" &&
-        typeof serverSettings.history.timelapse !== "undefined" &&
-        serverSettings.history.timelapse.deleteAfter
-      ) {
+      if (serverSettings?.history?.timelapse?.deleteAfter) {
```

Comment:
Cleaned up if check

Reply:
Nice, this is definitely a MongoDB/No-SQL fallacy. Have you considered writing database checks at startup to migrate specific versions of the database to a new working state?

Reply:
Yes, it needs to be done in another full PR for settings though. Thinking of tackling that before I return the history re-write as they should certainly be fixed and checked if not available. Also cached.

Reply:
Issue would be great. I did startup migrations in C# as well, have a lot of effective tactics and tips.

Reply:
#457 done :)